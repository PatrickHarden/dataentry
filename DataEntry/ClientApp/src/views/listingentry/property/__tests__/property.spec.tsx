import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Property from '../property';

describe('components', () => {
  describe('PropertyView', () => {
    let props: any;
    let wrapper: any;
    let store: any;

    const initialState = {
      system: {
        configDetails: {
          loaded: true,
          config: {
            featureFlags: {
              defaultWaterMarkTrue: false
            }
          }
        }
      },
      entry : {
        refreshImages: []
      },
      mapping: {
        country: {
        }
      }
    };
    const mockStore = configureStore();

    beforeEach(() => {
      props = {
        listing: {
          propertyType: {},
          walkthrough: ""
        },
        validations: {
          'field1': {required: true}
        }
      }
      store = mockStore(initialState);
    });

    it("check the existence of 3D space tag inside the property component", () => {
      wrapper = mount(<Provider store={store}><Property {...props} /></Provider>);
      expect(wrapper.exists('[name="walkThrough"]')).toBeTruthy()
    });

    it("check the existence of Building Display Name tag inside the property component", () => {
      wrapper = mount(<Provider store={store}><Property {...props} /></Provider>);
      expect(wrapper.exists('[name="propertyName"]')).toBeTruthy()
    });

    it("given a prop the value is set", () => {
      const walkthrough = "www.CBREWalkthrough.com";
      props.listing.walkThrough = walkthrough;
      wrapper = mount(<Provider store={store}><Property {...props} /></Provider>);
      expect(wrapper.exists('[id="walkThrough"][defaultValue="' + walkthrough + '"]')).toBeTruthy()
    });

    it("given a prop the value is set", () => {
      const propertyName = "Building Name";
      props.listing.propertyName = propertyName ;
      wrapper = mount(<Provider store={store}><Property {...props} /></Provider>);
      expect(wrapper.exists('[id="propertyName"][defaultValue="' + propertyName + '"]')).toBeTruthy()
    });

    // snapshot render
    it('renders property to snapshot', () => {
      expect(shallow(<Provider store={store}><Property {...props} /></Provider>)).toMatchSnapshot();
    });
  });
});