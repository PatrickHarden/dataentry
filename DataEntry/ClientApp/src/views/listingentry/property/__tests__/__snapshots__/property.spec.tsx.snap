// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components PropertyView renders property to snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Memo(Property)
    listing={
      Object {
        "propertyType": Object {},
        "walkthrough": "",
      }
    }
    validations={
      Object {
        "field1": Object {
          "required": true,
        },
      }
    }
  />
</ContextProvider>
`;
